File: apps/api/src/main/java/com/derbysoft/click/modules/identityaccess/domain/entities/TenantMembership.java
package com.derbysoft.click.modules.identityaccess.domain.entities;

import com.derbysoft.click.modules.identityaccess.domain.valueobjects.Role;
import java.time.Instant;
import java.util.UUID;

public record TenantMembership(
    UUID id,
    UUID userId,
    UUID tenantId,    // = PropertyGroup.id
    Role role,
    Instant createdAt
) {}

File: libs/domain/src/main/kotlin/domain/TenantMembership.kt
package domain

import java.time.Instant
import java.util.UUID

data class TenantMembership(
    val id: UUID,
    val userId: UUID,
    val tenantId: UUID,
    val role: Role,
    val createdAt: Instant,
    val updatedAt: Instant
)
